legalize-vadd.mir

```diff
@@ -0,0 +1,58 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
```

Comment:
yes, I will append them there, now it's just easier for me to test 

When I append the tests there, since we have [rv32](https://github.com/llvm/llvm-project/tree/main/llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv32) and [rv64](https://github.com/llvm/llvm-project/tree/main/llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv64), should I just keep all the test cases the same but just change `-mtriple=riscv32` and `-mtriple=riscv64`?

Reply:
I think you can put a copy of each test case in each file:
* [rv32/legalize-add.mir](https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv32/legalize-add.mir)
* [rv64/legalize-add.mir](https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv64/legalize-add.mir)

Reply:
[This](https://github.com/llvm/llvm-project/blob/e836283d43ef598180acb3e61d0c6259606ec85a/llvm/test/CodeGen/RISCV/GlobalISel/irtranslator/vec-args.ll) file should give you a good idea which types we'd like to legalize.

Reply:
> I think you can put a copy of each test case in each file:

sounds good!

This is irrelevant to my test cases, but is it even possible to pass `i7`, like in [here](https://github.com/llvm/llvm-project/blob/5aa934e2af8727852dec0ec1cfa0cba05d858f70/llvm/test/CodeGen/RISCV/GlobalISel/legalizer/rv32/legalize-add.mir#L5)?

Reply:
It is [allowed](https://llvm.org/docs/LangRef.html#integer-type)